physics/discrete_trajectory_body.hpp

```diff
@@ -168,7 +208,26 @@ void DiscreteTrajectory<Frame>::ForgetAfter(Instant const& time) {
   // entry and all the entries that follow it.  This preserves any entry with
   // time == |time|.
   auto const it = timeline_.upper_bound(time);
+  if (downsampling_.has_value()) {
+    if (it != timeline_.end() &&
+        downsampling_->start_of_dense_timeline->first >= it->first) {
+      // The start of the dense timeline will be invalidated.
+      if (it == timeline_.begin()) {
+        // The timeline will be empty after erasing.
+        downsampling_->start_of_dense_timeline = timeline_.end();
```

Comment:
Are we sure that `end` is not invalidated?

Reply:
This is what cppreference says: http://en.cppreference.com/w/cpp/container/map/erase.